policy-controller/core/src/outbound.rs

```diff
@@ -24,10 +24,16 @@ pub type HttpRoute = OutboundRoute<HttpRouteMatch, HttpRetryCondition>;
 pub type GrpcRoute = OutboundRoute<GrpcRouteMatch, GrpcRetryCondition>;
 pub type RouteSet<T> = HashMap<GroupKindNamespaceName, T>;
 
+pub enum TargetKind {
+    UnmeshedNetwork,
+    Service,
+}
+
 pub struct OutboundDiscoverTarget {
-    pub service_name: String,
-    pub service_namespace: String,
-    pub service_port: NonZeroU16,
+    pub kind: TargetKind,
+    pub name: String,
+    pub namespace: String,
+    pub port: NonZeroU16,
```

Comment:
moderately-held opinion: we should refrain from renaming other fields in the scope of this pr, which i'd hope could remain focused on _only_ adding a `TargetKind` to this outbound structure.

Reply:
Makes sense as a general rule. I would rather include this change here though. It is quite clear why we are renaming these in the context of this larger change.

Reply:
makes sense. resolved ✔️